tools/r2d2/custom.py

```diff
@@ -0,0 +1,148 @@
+import pandas as pd
+import sys
+import time
+
+prediction_value = None
+
+
+class R2D2:
+    MEGA = 10 ** 6
+    MEGA_STR = " " * MEGA
+
+    CMD_COL = "cmd"
+    ARG_COL = "arg"
+
+    def __init__(self):
+        self._mem_array = []
+        print("R2D2 __init__")
+
+    def _mem_size_mb(self):
+        return len(self._mem_array)
+
+    def _clear_memory(self):
+        self._mem_array = []
+
+    def _alloc_additional_memory(self, additional_mb):
+        for i in range(0, additional_mb):
+            try:
+                self._mem_array.append(R2D2.MEGA_STR + str(i))
+                print("Adding: {}".format(i))
```

Comment:
Remove this, otherwise it is printed thousands of times.

Reply:
This was exactly the point for printing it.. r2d2 is a debug tool.. you want to see what was the last memory allocation the went ok.. and to see the rate of memory allocations.

Reply:
But last memory allocation you can print in the MemoryError exception. Or in the end of the loop.

And you will not see rate of memory allocations when do this in pred server mode. Especially, i think, OS is smart enough to allocate BIG chunk if there were many requests for smaller chunks. So I don't see these printouts help you.

Reply:
I will print every 100MB of memory.. 
We don't see the exception.. the OOM killer or docker is just killing the entire process..